File: adm_apresentacao.py
"""Apresentação executiva — leitura agregada e sem dados pessoais.

Para que serve: permitir mostrar a plataforma a alguém de fora (um
investidor, por exemplo) em menos de um minuto, usando OS MESMOS dados do
sistema — nunca números de demonstração.

Duas garantias, ambas no servidor (esconder botão no navegador não é
proteção):

1. A rota vive sob ``/api/admin/``. Além da checagem explícita de
   ``autorizado()`` aqui, o ``before_request`` de main.py já barra todo
   ``/api/admin/*`` sem chave administrativa válida.

2. O corpo devolvido é montado campo a campo, só com escalares e
   enumerações fechadas. Nome, e-mail, telefone, endereço, identificador
   de contato e texto livre (``inferencia``/``motivo``, que pode citar uma
   empresa) NUNCA entram aqui. Nada de ``SELECT *``.

Este módulo não escreve, não envia, não executa e não cria link público.
Ele apenas lê o que já existe. Onde não há medição, devolve ``None`` — a
interface mostra a etapa real alcançada em vez de inventar um resultado.
"""
from datetime import datetime, timezone

from psycopg2.extras import RealDictCursor


# Enumerações fechadas: só estes valores podem sair daqui em campos de
# texto. Qualquer coisa fora da lista vira "outro".
PRIORIDADES = ('urgente', 'alta', 'normal', 'baixa')
ACOES_CONHECIDAS = (
    'pesquisar', 'classificar', 'analisar', 'planejar', 'sugerir_conteudo',
    'enviar_followup', 'avaliar_reengajamento', 'revisar_bloqueio',
    'decidir_proposta', 'priorizar_atendimento', 'informar_diretor',
)
ESTAGIOS_ABERTOS = ('novo', 'qualificacao', 'degustacao', 'proposta', 'negociacao')


def _enum(valor, permitidos):
    return valor if valor in permitidos else 'outro'


def _contagens_comerciais(factory):
    """Agregados puros: nenhuma linha individual sai desta função."""
    conn = factory()
    try:
        conn.set_session(readonly=True, isolation_level='REPEATABLE READ')
        with conn.cursor(cursor_factory=RealDictCursor) as cur:
            cur.execute("SET LOCAL statement_timeout='15s'")

            cur.execute(
                "SELECT estagio, count(*) AS total FROM leads_crm "
                "WHERE NOT cadastro_teste AND NOT arquivado GROUP BY estagio"
            )
            por_estagio = {linha['estagio']: linha['total'] for linha in cur.fetchall()}

            cur.execute(
                "SELECT count(DISTINCT (cidade, estado)) AS total FROM leads_crm "
                "WHERE NOT cadastro_teste AND NOT arquivado AND cidade IS NOT NULL"
            )
            pracas = cur.fetchone()['total']

            cur.execute("SELECT status, count(*) AS total FROM pedidos GROUP BY status")
            pedidos_por_status = {linha['status']: linha['total'] for linha in cur.fetchall()}

            cur.execute(
                "SELECT COALESCE(sum(valor_centavos),0) AS total FROM pedidos WHERE status='pago'"
            )
            receita_confirmada = cur.fetchone()['total']

        abertos = sum(por_estagio.get(e, 0) for e in ESTAGIOS_ABERTOS)
        return {
            'contatos_em_relacionamento': abertos,
            'contatos_por_estagio': {e: por_estagio.get(e, 0) for e in ESTAGIOS_ABERTOS},
            'clientes': por_estagio.get('cliente', 0),
            'pracas_alcancadas': pracas,
            'pedidos_registrados': sum(pedidos_por_status.values()),
            'pedidos_pagos': pedidos_por_status.get('pago', 0),
            'receita_confirmada_centavos': int(receita_confirmada or 0),
        }
    finally:
        conn.close()


def _contagens_propostas(factory):
    """Fila de ações comerciais: o coração do controle humano.

    A tabela pertence a acoes_comerciais.py (migration própria). Se ela
    ainda não existir no ambiente em uso, isso não é erro de apresentação
    — devolve None e a interface diz que a fila não foi lida.
    """
    conn = factory()
    try:
        conn.set_session(readonly=True, isolation_level='REPEATABLE READ')
        with conn.cursor(cursor_factory=RealDictCursor) as cur:
            cur.execute("SET LOCAL statement_timeout='10s'")
            cur.execute(
                "SELECT status, count(*) AS total FROM acoes_comerciais_propostas GROUP BY status"
            )
            por_status = {linha['status']: linha['total'] for linha in cur.fetchall()}
    except Exception:
        return None
    finally:
        conn.close()

    return {
        'propostas': sum(por_status.values()),
        'aguardando_aprovacao': por_status.get('aguardando_aprovacao', 0),
        'aprovadas': por_status.get('aprovada', 0) + por_status.get('executando', 0),
        'rejeitadas': por_status.get('rejeitada', 0),
        'executadas': por_status.get('enviada', 0),
        'sem_confirmacao': por_status.get('bloqueada', 0) + por_status.get('incerta', 0),
    }


def _oportunidade_anonima(leitura):
    """Uma oportunidade real, descrita só por campos de enumeração.

    ``inferencia``/``motivo`` ficam de fora de propósito: é texto livre e
    pode citar o nome de uma empresa ou de uma pessoa.
    """
    if not leitura:
        return None
    oportunidades = (leitura.get('comercial') or {}).get('oportunidades') or []
    if not oportunidades:
        return None
    escolhida = min(
        oportunidades,
        key=lambda o: PRIORIDADES.index(o['prioridade']) if o.get('prioridade') in PRIORIDADES else 99,
    )
    return {
        'tipo': _enum(escolhida.get('tipo_decisao'), ('oportunidade_parada',)),
        'prioridade': _enum(escolhida.get('prioridade'), PRIORIDADES),
        'proxima_acao': _enum(escolhida.get('proxima_acao'), ACOES_CONHECIDAS),
        'exige_aprovacao': bool(escolhida.get('exige_aprovacao', True)),
        'total_abertas': len(oportunidades),
    }


def _capacidades(factory):
    """O que está de fato funcionando hoje, canal a canal.

    Reaproveita canais_status.status_todos_os_canais — a mesma leitura que
    o ADM já mostra, sem nenhuma chamada de rede nova.
    """
    try:
        from canais_status import status_todos_os_canais
        canais = status_todos_os_canais(factory)
    except Exception:
        return None
    return [
        {
            'canal': canal['canal'],
            'estado': canal['estado'],
            'leitura': bool(canal['leitura_disponivel']),
            'escrita': bool(canal['escrita_disponivel']),
            'aprovacao_exigida': bool(canal['aprovacao_exigida']),
        }
        for canal in canais
    ]


def _etapa_alcancada(propostas, comercial):
    """Quando ainda não há resultado comprovado, dizer a etapa REAL.

    Nada de "em breve" nem de projeção: a frase descreve o ponto mais
    avançado que os próprios registros sustentam.
    """
    if propostas is None:
        return 'fila_nao_lida'
    if propostas['executadas'] > 0:
        return 'acao_executada'
    if propostas['aprovadas'] > 0:
        return 'acao_aprovada'
    if propostas['aguardando_aprovacao'] > 0:
        return 'acao_aguardando_aprovacao'
    if propostas['propostas'] > 0:
        return 'acao_proposta'
    if comercial and comercial['contatos_em_relacionamento'] > 0:
        return 'relacionamento_em_andamento'
    return 'operacao_preparada'


def montar_apresentacao(factory, agora=None):
    """Cada bloco é isolado: a falha de um não derruba a apresentação
    inteira — vira ``None`` e a interface explica o que não foi lido."""
    agora = agora or datetime.now(timezone.utc)

    try:
        comercial = _contagens_comerciais(factory)
    except Exception:
        comercial = None

    propostas = None
    try:
        propostas = _contagens_propostas(factory)
    except Exception:
        propostas = None

    leitura = None
    try:
        from mi_diretor import leitura_diretor
        leitura = leitura_diretor(factory)
    except Exception:
        leitura = None

    conselho = (leitura or {}).get('conselho') or {}
    resultados = conselho.get('resultados_recentes') or []

    return {
        'gerado_em': agora.isoformat(),
        'negocio': {
            'marca': 'Maranhão Cordial',
            'categoria': 'Cordiais brasileiros sem álcool',
        },
        'situacao': comercial,
        'oportunidade': _oportunidade_anonima(leitura),
        'trabalho_da_tecnologia': None if leitura is None else {
            'em_atividade': bool(leitura.get('ia_trabalhando')),
            'concluidas_recentes': len((leitura.get('hoje') or {}).get('concluido') or []),
            'planejadas_hoje': len((leitura.get('hoje') or {}).get('planejado') or []),
            'aguardando_decisao_humana': len((leitura.get('hoje') or {}).get('precisa_de_mim') or []),
            'especialistas_ativos': int(conselho.get('trabalhando') or 0),
            'divergencias_registradas': len(conselho.get('conflitos') or []),
            'vetos_registrados': len(conselho.get('vetos') or []),
        },
        'controle_humano': None if propostas is None else dict(
            propostas,
            envio_automatico=False,
            aprovacao_exigida=True,
        ),
        'resultado': {
            'confirmados': len(resultados),
            'etapa_alcancada': _etapa_alcancada(propostas, comercial),
            'sem_confirmacao': None if propostas is None else propostas['sem_confirmacao'],
        },
        'capacidades': _capacidades(factory),
        'limites': [
            'Receita considera apenas pedidos com pagamento confirmado.',
            'Oportunidades e propostas não são receita.',
            'Nenhum número de economia de tempo, retorno ou conversão é estimado: só o que está medido aparece.',
            'Esta visão é somente leitura e não expõe contatos, nomes ou dados pessoais.',
        ],
    }


def nada_foi_lido(dados):
    """True quando nenhuma das fontes respondeu.

    Degradar bloco a bloco é o comportamento desejado (uma fonte fora do ar
    não derruba a leitura inteira). Mas se NADA foi lido não existe
    apresentação: a rota responde 503 em vez de devolver uma tela toda
    vazia como se fosse o estado real do negócio.
    """
    # 'capacidades' fica de fora: ela é lida de variáveis de ambiente e
    # responde mesmo com o banco inteiro fora do ar. Sozinha, não é uma
    # apresentação do negócio.
    return all(
        dados.get(bloco) is None
        for bloco in ('situacao', 'controle_humano', 'trabalho_da_tecnologia')
    )


def registrar_rotas_adm_apresentacao(app, factory, autorizado):
    from flask import jsonify

    @app.route('/api/admin/apresentacao', methods=['GET'])
    def adm_apresentacao():
        # Dupla checagem proposital: a rota não depende só do
        # before_request para existir de forma segura.
        if not autorizado():
            return jsonify(success=False, error='Não autorizado.'), 401
        try:
            dados = montar_apresentacao(factory)
            if nada_foi_lido(dados):
                return jsonify(success=False, error='Apresentação indisponível.'), 503
            return jsonify(success=True, apresentacao=dados)
        except Exception:
            app.logger.exception('Falha ao montar a apresentação executiva')
            return jsonify(success=False, error='Apresentação indisponível.'), 503

File: maranhao-backend/adm-apresentacao.js
/* ============================================================
   APRESENTAÇÃO EXECUTIVA — somente leitura, quatro passos.

   Usa os mesmos dados e a mesma identidade do sistema: uma única
   chamada GET a /api/admin/apresentacao, que já devolve o corpo
   agregado e sem dados pessoais (adm_apresentacao.py). Nenhuma
   escrita, nenhum envio, nenhum link público é criado aqui.

   A restrição de acesso é do servidor: /api/admin/* exige chave
   administrativa válida. Esta tela apenas desenha o que a rota
   autorizou — se a chave não existir, não há o que mostrar.

   Tradução: todo texto visível vive em TEXTOS[idioma]. Acrescentar
   um idioma é acrescentar uma chave — nenhuma frase está presa ao
   desenho. Nenhum idioma adicional é escolhido aqui: a preferência
   é confirmada antes de ser disponibilizada.
   ============================================================ */

(() => {
    'use strict';

    const TEXTOS = {
        'pt-BR': {
            selo: 'Apresentação executiva',
            fechar: 'Fechar',
            somenteLeitura: 'Somente leitura',
            passos: ['O negócio', 'A oportunidade', 'O controle', 'O resultado'],
            carregando: 'Preparando a leitura…',
            indisponivel: 'A apresentação não pôde ser montada agora.',
            indisponivelNota: 'Nenhum dado foi alterado. Tente novamente em instantes.',
            semDado: 'sem dados',
            negocio: {
                eyebrow: 'Maranhão Cordial',
                titulo: 'Cordiais brasileiros, com a operação registrada em um só lugar.',
                linha: 'A plataforma registra relacionamento, pedidos, canais e decisões no mesmo lugar. Os números abaixo são leitura direta dos registros — não há dados de demonstração.',
                contatos: 'Contatos em relacionamento',
                contatosNota: 'Estágios abertos, sem clientes fechados nem cadastros de teste.',
                pracas: 'Praças alcançadas',
                pracasNota: 'Cidades distintas com relacionamento registrado.',
                pedidos: 'Pedidos pagos',
                pedidosNota: 'Com pagamento confirmado; de {total} pedidos registrados.',
                receita: 'Receita confirmada',
                receitaNota: 'Somente pedidos pagos. Propostas e oportunidades não entram aqui.',
            },
            oportunidade: {
                eyebrow: 'Do sinal ao trabalho',
                titulo: 'Uma oportunidade real, com trabalho registrado sobre ela.',
                semOportunidade: 'Nenhuma oportunidade aberta está registrada neste momento.',
                semOportunidadeNota: 'A plataforma não cria oportunidades para ter o que mostrar. Quando houver sinal nos registros, ela aparece aqui.',
                prioridade: 'Prioridade atribuída',
                acao: 'Ação sugerida',
                aprovacao: 'Exige aprovação humana',
                abertas: 'Oportunidades abertas',
                prioridades: { urgente: 'Urgente', alta: 'Alta', normal: 'Normal', baixa: 'Baixa', outro: 'Não classificada' },
                acoes: {
                    pesquisar: 'Procurar contatos', classificar: 'Classificar',
                    analisar: 'Analisar', planejar: 'Planejar',
                    sugerir_conteudo: 'Sugerir conteúdo', enviar_followup: 'Enviar retorno',
                    avaliar_reengajamento: 'Avaliar reengajamento', revisar_bloqueio: 'Revisar bloqueio',
                    decidir_proposta: 'Decidir proposta', priorizar_atendimento: 'Priorizar atendimento',
                    informar_diretor: 'Informar a direção', outro: 'Não classificada',
                },
                sim: 'sim', nao: 'não',
                trabalho: 'Trabalho da tecnologia',
                concluidas: 'Atividades concluídas',
                planejadas: 'Planejadas para hoje',
                especialistas: 'Especialistas em atividade',
                divergencias: 'Divergências registradas',
            },
            controle: {
                eyebrow: 'Controle humano',
                titulo: 'Nada sai daqui sem uma decisão de uma pessoa.',
                linha: 'Cada ação proposta pela tecnologia entra numa fila de aprovação. Aprovar registra a decisão; a execução é um passo separado e auditado. Não existe envio automático.',
                propostas: 'Ações propostas',
                aguardando: 'Aguardando aprovação',
                aprovadas: 'Aprovadas',
                rejeitadas: 'Rejeitadas',
                executadas: 'Executadas',
                filaNaoLida: 'A fila de aprovações não pôde ser lida agora.',
                envio: 'Envio automático: desligado.',
            },
            resultado: {
                eyebrow: 'Resultado',
                titulo: 'O que está comprovado — e o que ainda não está.',
                confirmados: 'Resultados confirmados',
                semConfirmacao: 'Execuções sem confirmação',
                etapa: 'Etapa alcançada',
                etapas: {
                    acao_executada: 'Ações aprovadas já foram executadas e auditadas.',
                    acao_aprovada: 'Há ações aprovadas aguardando execução.',
                    acao_aguardando_aprovacao: 'Há ações propostas aguardando decisão humana.',
                    acao_proposta: 'A tecnologia já propõe ações a partir dos registros.',
                    relacionamento_em_andamento: 'O relacionamento comercial está registrado e em andamento.',
                    operacao_preparada: 'A operação está preparada; ainda não há ação comercial proposta.',
                    fila_nao_lida: 'A fila de aprovações não pôde ser lida agora.',
                },
                capacidades: 'Capacidades em operação',
                estados: { conectado: 'Conectado', pendente: 'Pendente', bloqueado: 'Bloqueado' },
                leitura: 'leitura',
                escrita: 'escrita',
            },
            limites: 'Critérios desta leitura',
            traducao: 'Interface preparada para tradução. Um idioma adicional é habilitado depois que a preferência for confirmada.',
        },
    };

    function textos(idioma) {
        return TEXTOS[idioma] || TEXTOS['pt-BR'];
    }

    /* Formatação pura — exportada para teste. */
    function formatarDinheiro(centavos, idioma) {
        if (centavos === null || centavos === undefined) return null;
        return (Number(centavos) / 100).toLocaleString(idioma || 'pt-BR', {
            style: 'currency', currency: 'BRL', maximumFractionDigits: 0,
        });
    }

    /* Nenhum campo de texto livre é aceito: se o servidor mandasse algo
       fora da enumeração (não manda), a interface mostraria o código cru
       em vez de inventar uma frase. */
    function descreverEtapa(codigo, idioma) {
        const t = textos(idioma);
        return t.resultado.etapas[codigo] || codigo || '—';
    }

    if (typeof module !== 'undefined' && module.exports) {
        module.exports = { TEXTOS, textos, formatarDinheiro, descreverEtapa };
    }

    if (typeof document === 'undefined') return;

    const API = 'https://maranhao-cordial-api.onrender.com';
    let idioma = 'pt-BR';
    let dados = null;
    let passo = 0;
    let carregando = false;

    const criar = (tag, texto, classe) => {
        const no = document.createElement(tag);
        if (texto !== undefined && texto !== null) no.textContent = texto;
        if (classe) no.className = classe;
        return no;
    };
    const numero = v => (v === null || v === undefined ? null : Number(v).toLocaleString('pt-BR'));

    const painel = criar('div', undefined, 'adm-apresentacao');
    painel.hidden = true;
    painel.setAttribute('role', 'dialog');
    painel.setAttribute('aria-modal', 'true');

    const topo = criar('div', undefined, 'adm-ap-topo');
    const selo = criar('span', '', 'adm-ap-selo');
    const somenteLeitura = criar('span', '', 'adm-topbar-botao');
    somenteLeitura.style.pointerEvents = 'none';
    const passosBox = criar('div', undefined, 'adm-ap-passos');
    const fechar = criar('button', '', 'adm-topbar-botao');
    fechar.type = 'button';
    topo.append(selo, somenteLeitura, passosBox, fechar);

    const palco = criar('div', undefined, 'adm-ap-palco');
    painel.append(topo, palco);
    document.body.append(painel);

    function caixaNumero(rotulo, valor, nota) {
        const caixa = criar('article', undefined, 'adm-ap-num');
        caixa.append(criar('span', rotulo));
        caixa.append(criar('strong', valor === null ? textos(idioma).semDado : valor));
        if (nota) caixa.append(criar('small', nota));
        return caixa;
    }

    function desenharNegocio(t) {
        const s = dados.situacao;
        palco.append(
            criar('p', t.negocio.eyebrow, 'adm-ap-eyebrow'),
            criar('h2', t.negocio.titulo, 'adm-ap-titulo'),
            criar('p', t.negocio.linha, 'adm-ap-linha')
        );
        const grade = criar('div', undefined, 'adm-ap-grade');
        grade.append(
            caixaNumero(t.negocio.contatos, s ? numero(s.contatos_em_relacionamento) : null, t.negocio.contatosNota),
            caixaNumero(t.negocio.pracas, s ? numero(s.pracas_alcancadas) : null, t.negocio.pracasNota),
            caixaNumero(t.negocio.pedidos, s ? numero(s.pedidos_pagos) : null,
                s ? t.negocio.pedidosNota.replace('{total}', numero(s.pedidos_registrados)) : null),
            caixaNumero(t.negocio.receita, s ? formatarDinheiro(s.receita_confirmada_centavos, idioma) : null, t.negocio.receitaNota)
        );
        palco.append(grade);
    }

    function desenharOportunidade(t) {
        const o = dados.oportunidade;
        const trabalho = dados.trabalho_da_tecnologia;
        palco.append(
            criar('p', t.oportunidade.eyebrow, 'adm-ap-eyebrow'),
            criar('h2', t.oportunidade.titulo, 'adm-ap-titulo')
        );
        if (!o) {
            palco.append(
                criar('p', t.oportunidade.semOportunidade, 'adm-ap-linha'),
                criar('p', t.oportunidade.semOportunidadeNota, 'adm-ap-rodape')
            );
        } else {
            const grade = criar('div', undefined, 'adm-ap-grade');
            grade.append(
                caixaNumero(t.oportunidade.abertas, numero(o.total_abertas)),
                caixaNumero(t.oportunidade.prioridade, t.oportunidade.prioridades[o.prioridade] || o.prioridade),
                caixaNumero(t.oportunidade.acao, t.oportunidade.acoes[o.proxima_acao] || o.proxima_acao),
                caixaNumero(t.oportunidade.aprovacao, o.exige_aprovacao ? t.oportunidade.sim : t.oportunidade.nao)
            );
            palco.append(grade);
        }
        if (trabalho) {
            palco.append(criar('p', t.oportunidade.trabalho, 'adm-ap-eyebrow'));
            const grade = criar('div', undefined, 'adm-ap-grade');
            grade.append(
                caixaNumero(t.oportunidade.concluidas, numero(trabalho.concluidas_recentes)),
                caixaNumero(t.oportunidade.planejadas, numero(trabalho.planejadas_hoje)),
                caixaNumero(t.oportunidade.especialistas, numero(trabalho.especialistas_ativos)),
                caixaNumero(t.oportunidade.divergencias, numero(trabalho.divergencias_registradas))
            );
            palco.append(grade);
        }
    }

    function desenharControle(t) {
        const c = dados.controle_humano;
        palco.append(
            criar('p', t.controle.eyebrow, 'adm-ap-eyebrow'),
            criar('h2', t.controle.titulo, 'adm-ap-titulo'),
            criar('p', t.controle.linha, 'adm-ap-linha')
        );
        if (!c) {
            palco.append(criar('p', t.controle.filaNaoLida, 'adm-ap-rodape'));
            return;
        }
        const grade = criar('div', undefined, 'adm-ap-grade');
        grade.append(
            caixaNumero(t.controle.propostas, numero(c.propostas)),
            caixaNumero(t.controle.aguardando, numero(c.aguardando_aprovacao)),
            caixaNumero(t.controle.aprovadas, numero(c.aprovadas)),
            caixaNumero(t.controle.executadas, numero(c.executadas))
        );
        palco.append(grade, criar('p', t.controle.envio, 'adm-ap-rodape'));
    }

    function desenharResultado(t) {
        const r = dados.resultado;
        const capacidades = dados.capacidades || [];
        palco.append(
            criar('p', t.resultado.eyebrow, 'adm-ap-eyebrow'),
            criar('h2', t.resultado.titulo, 'adm-ap-titulo'),
            criar('p', descreverEtapa(r.etapa_alcancada, idioma), 'adm-ap-linha')
        );
        const grade = criar('div', undefined, 'adm-ap-grade');
        grade.append(
            caixaNumero(t.resultado.confirmados, numero(r.confirmados)),
            caixaNumero(t.resultado.semConfirmacao, numero(r.sem_confirmacao))
        );
        for (const capacidade of capacidades.slice(0, 6)) {
            const caixa = criar('article', undefined, 'adm-ap-num');
            caixa.append(criar('span', capacidade.canal));
            caixa.append(criar('strong', t.resultado.estados[capacidade.estado] || capacidade.estado));
            const partes = [];
            if (capacidade.leitura) partes.push(t.resultado.leitura);
            if (capacidade.escrita) partes.push(t.resultado.escrita);
            caixa.append(criar('small', partes.join(' · ') || '—'));
            grade.append(caixa);
        }
        palco.append(grade);
    }

    function desenhar() {
        const t = textos(idioma);
        selo.textContent = t.selo;
        somenteLeitura.textContent = t.somenteLeitura;
        fechar.textContent = t.fechar;

        passosBox.replaceChildren();
        t.passos.forEach((rotulo, indice) => {
            const botao = criar('button', rotulo, 'adm-ap-passo');
            botao.type = 'button';
            botao.setAttribute('aria-current', String(indice === passo));
            botao.addEventListener('click', () => { passo = indice; desenhar(); });
            passosBox.append(botao);
        });

        palco.replaceChildren();
        if (carregando) {
            palco.append(criar('p', t.carregando, 'adm-ap-linha'));
            return;
        }
        if (!dados) {
            palco.append(
                criar('h2', t.indisponivel, 'adm-ap-titulo'),
                criar('p', t.indisponivelNota, 'adm-ap-linha')
            );
            return;
        }

        if (passo === 0) desenharNegocio(t);
        else if (passo === 1) desenharOportunidade(t);
        else if (passo === 2) desenharControle(t);
        else desenharResultado(t);

        const rodape = criar('div', undefined, 'adm-ap-rodape');
        rodape.append(criar('strong', t.limites + ': '));
        rodape.append(document.createTextNode((dados.limites || []).join(' ')));
        rodape.append(criar('p', t.traducao));
        palco.append(rodape);
        palco.classList.remove('adm-entrando');
        void palco.offsetWidth;
        palco.classList.add('adm-entrando');
    }

    async function carregar() {
        carregando = true;
        desenhar();
        try {
            const resposta = await fetch(API + '/api/admin/apresentacao', {
                cache: 'no-store',
                headers: { 'X-Admin-Key': window.adminKeyAtual, 'Content-Type': 'application/json' },
            });
            const corpo = await resposta.json().catch(() => ({}));
            dados = (resposta.ok && corpo.success) ? corpo.apresentacao : null;
        } catch (erro) {
            dados = null;
        } finally {
            carregando = false;
            desenhar();
        }
    }

    function abrir() {
        if (!window.adminKeyAtual) return;
        painel.hidden = false;
        passo = 0;
        fechar.focus();
        carregar();
    }

    function fecharPainel() {
        painel.hidden = true;
    }

    fechar.addEventListener('click', fecharPainel);
    document.addEventListener('keydown', evento => {
        if (painel.hidden) return;
        if (evento.key === 'Escape') { fecharPainel(); return; }
        if (evento.key === 'ArrowRight') { passo = Math.min(3, passo + 1); desenhar(); }
        if (evento.key === 'ArrowLeft') { passo = Math.max(0, passo - 1); desenhar(); }
    });
    window.addEventListener('adm-abrir-apresentacao', abrir);
})();
